toolchain/check/handle_alias.cpp

```diff
@@ -0,0 +1,70 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/context.h"
+#include "toolchain/check/modifiers.h"
+#include "toolchain/parse/node_ids.h"
+#include "toolchain/sem_ir/ids.h"
+#include "toolchain/sem_ir/typed_insts.h"
+#include "toolchain/sem_ir/value_stores.h"
+
+namespace Carbon::Check {
+
+auto HandleAliasIntroducer(Context& context,
+                           Parse::AliasIntroducerId /*parse_node*/) -> bool {
+  context.decl_state_stack().Push(DeclState::Alias);
+  context.decl_name_stack().PushScopeAndStartName();
+  return true;
+}
+
+auto HandleAliasInitializer(Context& /*context*/,
+                            Parse::AliasInitializerId /*parse_node*/) -> bool {
+  return true;
+}
+
+auto HandleAlias(Context& context, Parse::AliasId /*parse_node*/) -> bool {
+  auto [expr_node, expr_id] = context.node_stack().PopExprWithParseNode();
+
+  auto name_context = context.decl_name_stack().FinishName();
+
+  LimitModifiersOnDecl(context, KeywordModifierSet::None,
+                       Lex::TokenKind::Namespace);
+  context.decl_state_stack().Pop(DeclState::Alias);
```

Comment:
I suspect we'll want to at least support access modifiers.
```suggestion
  LimitModifiersOnDecl(context, KeywordModifierSet::Access,
                       Lex::TokenKind::Namespace);
   if (!!(modifiers & KeywordModifierSet::Access)) {
    context.TODO(context.decl_state_stack().innermost().saw_access_modifier,
                 "access modifier");
  }
  context.decl_state_stack().Pop(DeclState::Alias);
```

Reply:
Done